odl/discr/discr_mappings.py

```diff
@@ -593,38 +553,57 @@ def __call__(self, xi, outp=None):
         if xi.ndim != 2:
             raise ValueError('`xi` has {} axes instead of 2.'.format(xi.ndim))
 
-        if xi.shape[1] != ndim:
+        if xi.shape[0] != ndim:
             raise ValueError('`xi` has axis 1 with length {} instead '
-                             'of the grid dimension {}.'.format(xi.shape[1],
+                             'of the grid dimension {}.'.format(xi.shape[0],
                                                                 ndim))
-        if outp is not None:
-            if not isinstance(outp, np.ndarray):
-                raise TypeError('`outp` {!r} not a `numpy.ndarray` '
-                                'instance.'.format(outp))
-            if outp.shape != (xi.shape[0],):
+        if out is not None:
+            if not isinstance(out, np.ndarray):
+                raise TypeError('`out` {!r} not a `numpy.ndarray` '
+                                'instance.'.format(out))
+            if out.shape != (xi.shape[0],):
                 raise ValueError('Output shape {} not equal to (n,), where '
                                  'n={} is the total number of evaluation '
-                                 'points.'.format(outp.shape, xi.shape[0]))
+                                 'points.'.format(out.shape, xi.shape[0]))
 
         xi = _ndim_coords_from_arrays(xi, ndim=ndim)
-        if xi.shape[-1] != ndim:
+        if xi.shape[0] != ndim:
             raise ValueError('The requested sample points xi have dimension '
                              '{}, but this _NearestInterpolator has '
-                             'dimension {}.'.format(xi.shape[-1], ndim))
-
-        indices, norm_distances = self._find_indices(xi.T)
-        return self._evaluate_nearest(indices, norm_distances, outp)
+                             'dimension {}.'.format(xi.shape[0], ndim))
+
+#        indices, norm_distances, out_of_bounds = self._find_indices(xi.T)
+#        if method == "linear":
+#            result = self._evaluate_linear(indices, norm_distances,
+#                                           out_of_bounds)
+#        elif method == "nearest":
+#            result = self._evaluate_nearest(indices, norm_distances,
+#                                            out_of_bounds)
+#        if not self.bounds_error and self.fill_value is not None:
+#            result[out_of_bounds] = self.fill_value
 
-    def _evaluate_nearest(self, indices, norm_distances, outp=None):
-        """Evaluate nearest interpolation. Modified for in-place."""
-        idx_res = []
-        for i, yi in zip(indices, norm_distances):
-            idx_res.append(np.where(yi <= .5, i, i + 1))
-        if outp is not None:
-            outp[:] = self.values[idx_res]
-            return outp
-        else:
-            return self.values[idx_res]
+        indices, norm_distances = self._find_indices(xi)
+        return self._evaluate(indices, norm_distances, out)
+
+#    def _find_indices(self, xi):
```

Comment:
I possibly need to adapt this function a bit depending on the type of discretization we have. I'll write a TODO instead.